src/Event.cc

```diff
@@ -375,7 +447,70 @@ PropertySet::Ptr Event::processTextMessage(cms::TextMessage* textMessage) {
 
     return unmarshall(text);
 }
+
+/** private method to try to add data to a property set
+ * \param typeInfo a string containing the name of the data type
+ * \param item a node of a boost::property_tree::ptree containing data
+ * \param key the name of the data
+ * \param ps a PropertySet to store the name and data into.
+ * \return true if data was added to the PropertySet
+  */
+bool Event::addDataItem(std::string typeInfo, boost::property_tree::ptree& item, std::string key, PropertySet& ps) {
+    if (typeInfo == "string") {
+        std::string value = item.get_value<std::string>();
+        ps.add(key, value);
+    } else if (typeInfo == "bool") {
```

Comment:
These are all the same, which makes me think there should be a template involved to remove the code duplication.